cadasta/gui/tools/wizard/step_project_creation03.py

```diff
@@ -196,6 +199,99 @@ def upload_locations(self):
                 tr('Finish with %d failed' % failed)
             )
 
+    def _url_post_parties(self):
+        """Get url to create a new party.
+
+        :returns: api url
+        :rtype: str, None
+        """
+        organisation = self.data['organisation']['slug']
+
+        if self.project_upload_result:
+            project = self.project_upload_result['slug']
+            return '/api/v1/organizations/%s/projects/%s/parties/' % (
+                organisation,
+                project
+            )
+        else:
+            return None
+
+    def _connect_post(self, network, post_data):
+        """Call post method.
+
+        :param network: Network connector
+        :type network: NetworkMixin
+
+        :param post_data: data to post
+        :type post_data: QByteArray
+
+        :returns: Tuple of post status and results
+        :rtype: ( bool, str )
+        """
+        network.connect_post(post_data)
+        while not network.reply.isFinished():
+            QCoreApplication.processEvents()
+
+        if not network.error:
+            return True, network.results.data()
+        else:
+            return False, network.results.data()
+
+    def upload_parties(self):
+        """Upload party from this project."""
+        self.set_status(
+            tr('Uploading parties')
+        )
+
+        party = 0
+
+        # reset progress bar
+        current_progress = 0
+        self.set_progress_bar(current_progress)
+        total_layer = len(self.data['locations']['features'])
+        progress_block = 100 / total_layer
+
+        post_url = self._url_post_parties()
+
+        if not post_url:
+            # Project is not uploaded
+            return
+
+        network = NetworkMixin(get_url_instance() + post_url)
+
+        for layer in self.data['locations']['features']:
+            if layer['fields']['party_name'] and layer['fields']['party_type']:
+                post_data = QByteArray()
+                post_data.append('name=%s&' % layer['fields']['party_name'])
+                post_data.append('type=%s&' % layer['fields']['party_type'])
+
+                status, result = self._connect_post(network, post_data)
+
+                if status:
+                    party += 1
+                else:
+                    self.set_progress_bar(0)
+                    self.set_status(
+                        tr('Error: ') + result
+                    )
+            else:
+                self.set_status(
+                    tr('No party attributes found')
+                )
+            current_progress += progress_block
+            self.set_progress_bar(current_progress)
+
+        if party == 0:
+            self.set_status(
+                tr('Not uploading any party')
+            )
+        else:
+            self.set_status(
+                tr('Finished uploading ') + str(party) + tr(' party')
```

Comment:
We shouldn't use concatenation of strings. Especially with translation, it's very difficult to translate a sentence with singular/plural because the way we translate might be different in another language. One call to `tr` is better per sentence.
`tr('Finished uploading {party} party'.format(party))`